api/internal/localizer/locloader.go

```diff
@@ -0,0 +1,382 @@
+	if err != nil {
+		if err := fSys.RemoveAll(newDir.String()); err != nil {
+			log.Printf("%s", errors.WrapPrefixf(err, errCleanDst).Error())
+		}
+		return "", errors.WrapPrefixf(err, "unable to establish localize destination")
+	}
+
+	return newDir, nil
+}
+
+// pathSuffix returns path relative to prefix, where both are cleaned absolute paths
+// and prefix is a directory known to contain path
+func pathSuffix(prefix string, path string) string {
+	if prefix == string(filepath.Separator) || prefix == path {
+		return path[len(prefix):]
+	}
+	return path[len(prefix)+1:]
+}
+
+// Root returns the path to the current target directory
+func (ll *LocLoader) Root() string {
+	return ll.ldr.Root()
+}
+
+// Dst returns the path to the directory in destination that corresponds to Root
+func (ll *LocLoader) Dst() string {
+	return ll.dst
+}
+
+func toFSPath(urlPath string) string {
+	return filepath.Join(strings.Split(urlPath, "/")...)
+}
+
+func locFilePath(u *url.URL) string {
+	// raw github urls are the only type of file urls kustomize accepts;
+	// path consists of org, repo, version, path in repo
+	return filepath.Join(localizeDir, u.Hostname(), toFSPath(u.Path))
+}
+
+// Load returns the contents of path and its corresponding localize destination path information
+// if path is a file that can be localized. Otherwise, Load returns an error.
+func (ll *LocLoader) Load(path string) ([]byte, *LocPath, error) {
+	// checks in root, and thus in scope
+	content, err := ll.ldr.Load(path)
+	if err != nil {
+		return nil, nil, errors.WrapPrefixf(err, "invalid file reference")
+	}
+	if loader.HasRemoteFileScheme(path) {
+		// parsable because prefixed by scheme
+		u, err := url.Parse(path)
+		if err != nil {
+			log.Fatalf(errors.WrapPrefixf(
+				err,
+				"%s: cannot parse validated file url '%s'", fatalPrefix, path).Error())
+		}
+		return content, &LocPath{true, locFilePath(u)}, nil
+	}
+	if !filepath.IsAbs(path) {
+		abs := filepath.Join(ll.ldr.Root(), path)
+		// avoid symlinks; only write file corresponding to actual location in root
+		// avoid path that Load() shows to be in root, but may traverse outside
+		// temporarily; for example, ../root/config; problematic for rename and
+		// relocation
+		dir, f, err := ll.fSys.CleanedAbs(abs)
+		if err != nil {
+			log.Fatalf(errors.WrapPrefixf(
+				err,
+				"%s: cannot clean validated file path '%s'", fatalPrefix, abs).Error())
+		}
+
+		cleanPath := dir.Join(f)
+		// target cannot reference newDir, as this load would've failed prior to localize;
+		// not a problem if remote because then reference could only be in newDir if repo copy,
+		// which will be cleaned, is inside newDir
+		if ll.local && dir.HasPrefix(ll.args.newDir) {
+			return nil, nil, errors.WrapPrefixf(errDstRef, errReference, cleanPath)
+		}
+		return content, &LocPath{false, pathSuffix(ll.ldr.Root(), cleanPath)}, nil
+	}
+
+	return nil, nil, errors.WrapPrefixf(errors.Errorf(errAlpha), "path '%s' is absolute", path)
```

Comment:
This can just be

```
return nil, nil fmt.Errorf("absolute paths are not yet supported: path %q is absolute", path)
```

I mentioned in an earlier comment that I'm pretty sure we remove the "static error check" in the linter

Reply:
Noted. I've also replaced other similarly wrapped error messages with `errors.Errorf`.